.pre-commit-config.yaml

```diff
@@ -1,25 +1,66 @@
+default_stages: [pre-commit, pre-push]
+default_install_hook_types: [pre-commit, pre-push]
 repos:
 - repo: https://github.com/gitleaks/gitleaks
   rev: v8.21.1
   hooks:
   - id: gitleaks
+    stages: [pre-commit]
 - repo: https://github.com/koalaman/shellcheck-precommit
   rev: v0.10.0
   hooks:
   - id: shellcheck
     args: ["--external-sources"]
+    stages: [pre-commit]
 - repo: https://github.com/pre-commit/pre-commit-hooks
   rev: v5.0.0
   hooks:
   - id: end-of-file-fixer
+    stages: [pre-commit]
   - id: trailing-whitespace
+    stages: [pre-commit]
   - id: detect-private-key
+    stages: [pre-commit]
 - repo: https://github.com/pylint-dev/pylint
   rev: v3.3.1
   hooks:
   - id: pylint
     exclude: ^hack/boilerplate/boilerplate.py$
+    stages: [pre-commit]
 - repo: https://github.com/jumanjihouse/pre-commit-hooks
   rev: 3.0.0
   hooks:
-    - id: forbid-binary
+  - id: forbid-binary
+    stages: [pre-commit]
+- repo: local
+  hooks:
+  - id: make-modules
+    name: Run make verify-modules
+    entry: make verify-modules
+    language: system
+    stages: [pre-push]
+  - id: make-gen
+    name: Run make verify-gen
+    entry: make verify-gen
+    language: system
+    stages: [pre-push]
+  - id: make-spellcheck
+    name: Run make verify-shellcheck
+    entry: make verify-shellcheck
+    language: system
+    stages: [ pre-push ]
```

Comment:
I am not sure if `[ pre-push ]` is any different to `[pre-push]` (extra space padding being added to `pre-push` string). 

Could please paste sample output of using this `.pre-commit.config` in the PR description or as a comment? 

---- 

This looks great to me btw!
/lgtm

Reply:
I added some example output to the PR description